### PR TITLE
Request validation

### PR DESCRIPTION
**Pull request by markburns**

### CODE REVIEW

File: lib/slanger/connection.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ def error e
     end
 
     def establish
-      @socket_id = SecureRandom.uuid
+      @socket_id = "%d.%d" % [Process.pid, SecureRandom.random_number(10 ** 6)]
```
Comment (stevegraham) [inline lib/slanger/connection.rb]: ?

**Comment by stevegraham** [comment]
:+1:

File: lib/slanger/connection.rb (hunk 1 above)
Comment (markburns) [inline lib/slanger/connection.rb]: Thanks to the pusher guys, cc @zimbatm it follows their format. It doesn't make a difference to us, but we can use the same validation method and it is probably less confusing if any switches between slanger/pusher

**Comment by markburns** [comment]
Going to go through the advisory before merging this. Not 100% sure I've got everything here, and could probably expand upon the test cases

**Comment by markburns** [comment]
So I think this covers things. Hopefully, @homakov will be able to chime in. I couldn't replicate the attack myself

**Comment by markburns** [comment]
I've bumped the version, @stevegraham can you push a release?

**Comment by stevegraham** [comment]
@markburns I've added you as an owner on RubyGems.

**Pull request merged**